src/__tests__/element-queries.js

```diff
@@ -244,8 +244,38 @@ test('can find any form control when label text is inside other elements', () =>
     </label>
   `)
 
-  const result = getAllByLabelText('Test Label')
-  expect(result).toHaveLength(7)
+  const nodeTypes = [
+    'button',
+    'input',
+    'meter',
+    'output',
+    'progress',
+    'select',
+    'textarea',
+  ]
+  nodeTypes.forEach(nodeType => {
+    expect(getByLabelText('Test Label', {selector: nodeType}).nodeName).toEqual(
+      nodeType.toUpperCase(),
+    )
+  })
+})
+
+test('returns the first form control inside a label', () => {
```

Comment:
I'm fine with this change. In cases like this it might be good to link to the spec that describes this behavior, either in the source or test.

Reply:
Updated 👍